src/prod/src/Federation/README.md

```diff
@@ -0,0 +1,3 @@
+# Federation subsystem
+
+The **Federation subsystem** uses the communication primitives provided by the [Transport subsystem](/src/prod/src/Transport#transport-subsystem) and stitches the various nodes into a single unified cluster that it can reason about. It provides the distributed systems primitives needed by the other subsystems - failure detection, [leader election](https://github.com/Microsoft/service-fabric/blob/f258f7579af9643dac6b1c75c93db9a3bcd28fdd/src/prod/src/Federation/VoterStore.cpp), and [consistent routing](https://github.com/Microsoft/service-fabric/blob/master/src/prod/src/Federation/RoutingManager.cpp). The federation subsystem is built on top of distributed hash tables with a 128-bit token space. The subsystem creates a [ring topology over the nodes](https://github.com/Microsoft/service-fabric/blob/master/src/prod/src/Federation/NodeRing.cpp), with each node in the ring being allocated a subset of the token space for ownership. For failure detection, the layer uses a leasing mechanism based on heart beating and arbitration. The federation subsystem also guarantees through intricate [join](https://github.com/Microsoft/service-fabric/blob/master/src/prod/src/Federation/JoinManager.cpp) and [departure]() protocols that only a single owner of a token exists at any time. This provides leader election and consistent routing guarantees. 
```

Comment:
also pointing to a specific commit instead of master.  Use relative links to solve